### PR TITLE
Send INVALID_DATA in case SubMenuIcon is invalid

### PR DESCRIPTION
**Pull request by VVeremjova**
Related to #953
After questions the requirement was clarified

### CODE REVIEW

**Comment by VVeremjova** [comment]
Please review @dtrunov

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 1]
```diff
@@ -95,8 +95,12 @@ void AddSubMenuRequest::Run() {
       (*message_)[strings::msg_params][strings::menu_name];
   msg_params[strings::app_id] = app->app_id();
 
-  if (((*message_)[strings::msg_params].keyExists(strings::sub_menu_icon)) &&
-      CheckSubMenuIcon()) {
+  if (((*message_)[strings::msg_params].keyExists(strings::sub_menu_icon))) {
+    if (!CheckSubMenuIcon()) {
+      LOG4CXX_ERROR(logger_, "Sub-menu icon is not valid.");
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @VVeremjova CheckSubMenuIcon contain redundant checking (*message_)[strings::msg_params].keyExists(strings::sub_menu_icon) please remove it.

Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @dtrunov See  1e0e153

Comment (LevchenkoS) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @VVeremjova, why do you decide to move check for `sub_menu_icon` existence out from `CheckSubMenuIcon()` function?

**Comment by VVeremjova** [comment]
Rebuild required

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc (hunk 1 above)
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @LevchenkoS If icon does not exist then it is ok. If icon exists and invalid then we should send INVALID_DATA.

**Pull request merged**